tenacity/__init__.py

```diff
@@ -396,11 +398,15 @@ def __iter__(self):
             else:
                 break
 
+    @abstractmethod
+    def __call__(self, *args, **kwargs):
+        pass
+
 
 class Retrying(BaseRetrying):
     """Retrying controller."""
 
-    def call(self, fn, *args, **kwargs):
+    def __call__(self, fn, *args, **kwargs):
```

Comment:
This rename might break the API, right?

Reply:
Well, I didn't find in any part of the documentation neither the API docs (see here https://tenacity.readthedocs.io/en/latest/api.html?highlight=Retrying#tenacity.Retrying) where the method `call` is mentioned. 

But nothing prevents that somebody noticed that the method exists and is public, so somebody could be using it. In fact, in my code, I'm using it as a workaround. 

I don't know which is the tenacity policy regarding that. 

I could do something like the following for `Retrying` class only:

```py
  def call(self, *args, **kwargs):
    """Deprecated. Use ``__call__`` instead"""
    warnings.warn("'call' function is deprecated. Use '__call__' instead")
    self.__call__(*args, **kwargs)
```

This would keep backwards compatibility. @jd what do you think?

Reply:
Works for me 👍

Reply:
@jd implemented here: https://github.com/jd/tenacity/pull/253/commits/a5830a952a93db79acc7a4394f881bec12f2e316 and here: https://github.com/jd/tenacity/pull/253/commits/5245c747e7e865a1ecc3e8319193e417dc7da4d1

Let me know what do you think.

Reply:
A little late to the party given this PR is merged, but should this raise a `DeprecationWarning` (rather than the default `UserWarning`) to match other parts of tenacity?